### PR TITLE
Add logo in support of Ukraine

### PR DESCRIPTION
**Pull request by Jordanlelay**

### CODE REVIEW

File: typescript/web/public/static/img/ukr.svg [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
...QDIgBwYAYEAyIAcGAGBAMiAHBgBgQA4IBMSAYEAOCATEgGBADggExIBgQA4IBMSAYEAOCATEgGBADggExIBgQA4IBMSAYEAOCATEgBgQDYkAwIAYEA2JAMCAGBANiQDAgBgQDYkAwIAYEA2JAMCAGBANiQDAgBgQDYkAwIAYEA2JAMCAGxIBgQAwIBsSAYEAMCAbEgGBADAgGxIBgQAwIBsSAYEAMCAbEgGBADAgGxIBgQAwIBsSAYEAMCAbEgBgQDIgBwYAYEAyIAcGAGBAMiAHBgBgQDIgBwYAYEAyIAcGAGBAMiAHBgBgQDIgBwYAYEAyIATEgGBADggExIBgQA4IBMSAYEAOCATEgGBADggExIBgQA4IBMSAYEAOCATEgGBADggExIBgQA4IBMSAGBANiQDAgBgQDYkAwIAYEA2JAMCAGBANiQDAgBgQDYkAwIAYEA2JAMCAGBANiQDAgBgQDYkAMKAEGxIBgQAwIBsSAYEAMCAbEgGBADAgGxIBgQAwIBsSAYEAMCAbEgGBADAgGxIBgQAwIBsSAYEAMiAHBgBgQDIgBwYAYEAyIAcGAGBAMiAHBgBgQDIgBwYAYEAyIAcGAGBAMiAHBgBgQDIgBwYAYEAyIATEgGBADggExIBgQA4IBMSAYEAOCATEgGBADggExIBgQA4IBMSAYEAOCATEgGBADggExIBgQA2JAMCAGBANiQDAgBgQDYkAwIAYEA2JAMCAGBANiQDAgBgQDYkAwIAYEA2JAMCAGBANiQDAgBgQDYkAMCAbEgGBADAgGxIBgQAwIBsSAYEAMCAbEgGBADAgGxIBgQAwIBsSAYEAMCAbEgGBADAgGxIBgQAyIAcGAGBAMiAHBgBgQDIgBwYAYEAyIAcGAGBAMiAHBgBgQDIgBwYAYEAyIAcGAGBAMiAHBgBgQA4IBMSAYEAOCATEgGBADggExIBgQA4IBMSAYEAOCATEgGBADggExIBgQA4IBMSAYEAOCATEgGBADYkAwIAYEA2JAMCAGBANiQDAgBgQD8skAm6xEiJfGzGsAAAAldEVYdGRhdGU6Y3JlYXRlADIwMjItMDItMjNUMTQ6NTk6MzIrMDA6MDBiaJcUAAAAJXRFWHRkYXRlOm1vZGlmeQAyMDIyLTAyLTIzVDE0OjU5OjMyKzAwOjAwEzUvqAAAAABJRU5ErkJggg=="/>
```
Comment (bbenoist) [inline typescript/web/public/static/img/ukr.svg:16]: @nicOwlas The "tag" icon is in Base64 in the full logo SVG but it's an actual SVG path in the icon-only version.
Isn't it because you've imported the icon as PNG in the full-version? Can you give us a version with no Base64 at-all?

Also, speaking about it, The `fill="black"` properties of each `<path />` tag in the file cab be removed in-favor of replacing the `fill="none"` property of the top-level `<svg /> tag to `fill="black"`.

Comment (nicOwlas) [reply typescript/web/public/static/img/ukr.svg:16]: Yes I added the flag as an image to fill the icon, I will remove the image and do the colors manually.

Comment (nicOwlas) [reply typescript/web/public/static/img/ukr.svg:16]: You can try those ones?
logo full ukraine
icon ukraine

File: typescript/web/src/components/auth/sign-in.tsx [hunk 2]
```diff
@@ -6,8 +6,8 @@ import {
 } from "@chakra-ui/react";
 import NextLink from "next/link";
 import React from "react";
-import { Logo } from "../logo";
 import { TextLink } from "../text-link";
+import { UkrainianLogo } from "../website/ukrainian-logo";
```
Comment (bbenoist) [inline typescript/web/src/components/auth/sign-in.tsx:10]: Can we avoid referencing the specific `UkranianLogo` in every page and just do the switch in the correct `Logo` component instead?

File: typescript/web/src/components/logo/logo.tsx [hunk 3]
```diff
@@ -9,22 +9,65 @@ export const Logo = React.forwardRef<
...
+          d="M260.21 125.126H304.094V144.56H235V39.7584H260.21V125.126Z"
           fill={useColorModeValue("black", "white")}
         />
+        <path
+          d="M357.388 144.56C356.526 142.977 355.76 140.65 355.089 137.578C350.636 143.193 344.412 146 336.415 146C329.089 146 322.864 143.793 317.741 139.378C312.618 134.915 310.056 129.325 310.056 122.607C310.056 114.161 313.168 107.779 319.393 103.46C325.618 99.1414 334.668 96.982 346.542 96.982H354.012V92.8792C354.012 85.7292 350.924 82.1542 344.747 82.1542C339.001 82.1542 336.128 84.9854 336.128 90.6478H311.923C311.923 83.114 315.107 76.9957 321.476 72.2931C327.892 67.5904 336.056 65.2391 345.968 65.2391C355.879 65.2391 363.708 67.6624 369.454 72.509C375.2 77.3556 378.145 84.0017 378.288 92.4473V126.925C378.384 134.075 379.486 139.546 381.592 143.337V144.56H357.388ZM342.233 128.725C345.25 128.725 347.739 128.077 349.703 126.781C351.714 125.486 353.15 124.022 354.012 122.391V109.938H346.973C338.546 109.938 334.332 113.729 334.332 121.311C334.332 123.518 335.075 125.318 336.559 126.71C338.043 128.053 339.935 128.725 342.233 128.725Z"
+          fill={useColorModeValue("black", "white")}
```
Comment (bbenoist) [inline typescript/web/src/components/logo/logo.tsx:30]: Can we use a variable instead of repeating calls to the hook?

File: typescript/web/src/components/logo/logo.tsx [hunk 4]
```diff
@@ -9,22 +9,65 @@ export const Logo = React.forwardRef<
   const { iconColor = "brand", logoOnly, ...rest } = props;
   return (
     <chakra.svg
-      aria-hidden
-      viewBox={logoOnly ? "0 0 84 84" : "0 0 393 84"}
+      viewBox={logoOnly ? "0 0 188 191" : "0 0 897 191"}
```
Comment (bbenoist) [inline typescript/web/src/components/logo/logo.tsx:12]: While on one side you've created an `UkranianLogo` component, on the other you've replaced some hardcoded values of the regular Logo component.

In-order to simplify the rollback even if we change some code based on the logo, can you please either:
* keep the official logo as `OfficialLogo` and the new one as `UkranianLogo` (and use the correct one in a `Logo` component)
or
* Don't change the original viewBox dimensions and use an SVG of the exact same dimensions than the official one.

File: typescript/web/src/components/auth/sign-in.tsx [hunk 5]
```diff
@@ -16,7 +16,13 @@ import { useSignIn } from "./sign-in.context";
 
 const BigLogo = () => (
   <NextLink href="/">
-    <Logo h="9" mb="10" iconColor="brand.600" mx="auto" cursor="pointer" />
+    <UkrainianLogo
+      h="9"
+      iconColor="brand.400"
```
Comment (bbenoist) [inline typescript/web/src/components/auth/sign-in.tsx:21]: Why has the iconColor changed? I thinked it was useless?

File: typescript/web/src/components/website/Navbar/NavContent.tsx [hunk 6]
```diff
@@ -70,7 +70,7 @@ const MobileNavContext = (props: FlexProps) => {
         </Box>
         <NextLink href="/website">
           <Box as="a" rel="home" mx="auto" cursor="pointer">
-            <Logo h="24px" iconColor="brand.400" />
+            <UkrainianLogo h="6" iconColor="brand.400" cursor="pointer" />
```
Comment (bbenoist) [inline typescript/web/src/components/website/Navbar/NavContent.tsx:73]: The cursor is already set with the Box above. Why did you need do add it again?

File: typescript/web/src/components/website/ukrainian-logo.tsx [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+import * as React from "react";
+
+export const UkrainianLogo = React.forwardRef<
+  SVGSVGElement,
+  HTMLChakraProps<"svg"> & { iconColor?: string }
+>((props: { iconColor?: string }, ref) => {
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const { iconColor = "brand", ...rest } = props;
+  return (
+    <chakra.svg
```
Comment (bbenoist) [inline typescript/web/src/components/website/ukrainian-logo.tsx:11]: Except from the viewbox ternary, it looks like this file is now exactly the same than the other Logo. Can we merge both?

**Review by bbenoist — CHANGES_REQUESTED** [review]

File: typescript/web/src/components/logo/logo.tsx (hunk 4 above)
Comment (bbenoist) [reply typescript/web/src/components/logo/logo.tsx:13]: @Jordanlelay You've _self-resolved_ this comment but there is only 1 logo SVG in the code (i.e. the official version has been removed) and the viewBox is different than what is was with the original version. Am I missing something?

What will we have to do once we'll have to go back to the official version? Will it be doable in ~5mn?

Comment (Jordanlelay) [reply typescript/web/src/components/logo/logo.tsx:13]: You asked me to merge both logos file, right ? So now there is only: the Ukrainian one.
Is there some problem using a `git revert` with the work I've done ?
Even manually, it should not take more than 10 minutes to go back to the original logo

Comment (bbenoist) [reply typescript/web/src/components/logo/logo.tsx:13]: > You asked me to merge both logos file, right ?

Not exactly:

> keep the official logo as OfficialLogo and the new one as UkranianLogo (and use the correct one in a Logo component)

Which means that we'd have 3 components:
* **_OfficialLogo_** - The official logo
* **_UkranianLogo_** - The Ukranian version
* _**Logo**_ - A component which just exposes either OfficialLogo or UkranianLogo component depending on what we'd like to display for now. I've not specified more details to let you free of doing the change as quick as you believe but we could simply switch between one version or another depending on a `NEXT_PUBLIC_APP_LOGO=ukranian` env variable.

Contrary to a git revert, this would prevent possible conflicts and bugs as we don't have any idea about how much time we'll keep the temporary logo and how much the code will evolve until then.

Comment (bbenoist) [reply typescript/web/src/components/logo/logo.tsx:13]: Anyway, as the PR is already taking a lot of time and risks of issues are limited, we can stay with the implementation you've done if you feel like it'd take too much time for you to do the changes 🚀

**Review by bbenoist — APPROVED** [review]
